test/Base_Tests/src/Runtime/Stack_Size_Spec.enso

```diff
@@ -0,0 +1,133 @@
+            msg = "Cannot find " + prefix + "* in " + dir.to_text + '\n'
+            err = dir.list.fold msg t-> f->
+                t + f.to_text + '\n'
+            Panic.throw err
+
+    project_root = File.new enso_project.root.to_text
+    repository_root = project_root . parent . parent
+    built_distribution = find_prefix repository_root "built-distribution"
+    enso_engine = find_prefix built_distribution "enso-engine-"
+    enso = find_prefix enso_engine "enso-"
+    bin = find_prefix enso "bin"
+
+    exe = File.new bin / if Platform.os == Platform.OS.Windows then "enso.bat" else "enso"
+
+    if exe.is_regular_file.not then Panic.throw "Cannot find "+exe.to_text
+
+    exe
+
+## Generates code for mapping over a vector with the given nesting level.
+   Returns code of the main method that is meant to be pasted into a separate module.
+   The code prints the count of Java frames to stdout in the deepest `Vector.map` call.
+
+   Example of the code is (for nesting_level 2):
+   ```
+   main =
+       vec = [[42]]
+       vec.map e0->
+           e0.map e1->
+               cnt = RuntimeException.new.getStackTrace.length
+               IO.println 'java_stack_frames='+cnt.to_text
+   ```
+
+   Arguments:
+   - nesting_level How many times should the vector be nested
+generate_code nesting_level:Integer -> Text =
+    bldr = Vector.Builder.new
+    bldr.append "from Standard.Base import all"
+    bldr.append "polyglot java import java.lang.RuntimeException"
+    bldr.append '\n'
+    bldr.append <| "main = "
+    bldr.append <| "    "
+                   + "vec = "
+                   + ("[" * nesting_level)
+                   + "42"
+                   + ("]" * nesting_level)
+    bldr.append <| "    "
+                   + "vec.map e0->"
+    0.up_to (nesting_level - 1) . each \i ->
+        bldr.append <| ("    " * (i + 2))
+                       + "e"
+                       + i.to_text
+                       + ".map e"
+                       + (i + 1).to_text
+                       + "-> "
+    bldr.append <| ("    " * (nesting_level + 1))
+                   + "cnt = RuntimeException.new.getStackTrace.length"
+    bldr.append <| ("    " * (nesting_level + 1))
+                   + "IO.println <| 'java_stack_frames=' + cnt.to_text"
+                   + '\n'
+    bldr.to_vector.reduce \first_line:Text second_line:Text ->
+        first_line + '\n' + second_line
+
+
+## Runs Enso subprocess with disabled Truffle compiler, with
+   larger thread stack and also with larger stack trace element collected
+   (which is needed for `new RuntimeException().getStackTrace().length`)
+   as this value is by default set only to 1024.
+
+   The thread stack size is also set to a sufficiently large value
+   to ensure there is no StackOverflow.
+run_without_compiler enso_args:Vector -> Process_Result =
+    java_opts = "-Dpolyglot.engine.Compilation=false "
+                + "-XX:MaxJavaStackTraceDepth=18000 "
+                + "-Xms16M"
```

Comment:
Can we get back to the original, default value of `-Xms1M`? CCing @AdRiley

Reply:
Adam's PR reverted in 9c3d18fb3b8bceeb66662c603e0e9c1b90ecdd42. Note that the `"-Xms16M"` specified on this line is only for testing purposes.